packages/web-api-client/src/a11y-service-authentication-handler.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { AuthenticationContext, TokenResponse } from 'adal-node';
+import { RequestPromiseOptions } from 'request-promise';
+import { isNullOrUndefined } from 'util';
+
+export interface A11yServiceCredential {
+    clientId: string;
+    clientSecret: string;
+}
+
+export class A11yServiceAuthenticationHandler {
+    constructor(
+        private readonly credential: A11yServiceCredential,
+        private readonly authenticationContext: AuthenticationContext,
+        private readonly resource: string,
+    ) {}
+
+    public async getToken(): Promise<TokenResponse> {
+        return new Promise((resolve, reject) => {
+            this.authenticationContext.acquireTokenWithClientCredentials(
+                this.resource,
+                this.credential.clientId,
+                this.credential.clientSecret,
+                (err, tokenResponse) => {
+                    if (!isNullOrUndefined(err)) {
+                        reject(err);
+                    } else {
+                        resolve(tokenResponse as TokenResponse);
+                    }
+                },
+            );
+        });
+    }
+
+    public async getAuthHeaders(): Promise<RequestPromiseOptions> {
+        const accessToken = await this.getToken();
```

Comment:
check if the sdk supports caching the token. if not then u need to cache token & get token again on expiry.

Reply:
They are caching tokens: see line 136 here: https://github.com/AzureAD/azure-activedirectory-library-for-nodejs/blob/889b2babaca7fe192f7c0c651526a4adc53c38ea/lib/token-request.js